src/graphics/program-lib/chunks/TBNObjectSpace.frag

```diff
@@ -0,0 +1,7 @@
+void getTBN() {
+    vec3 B = cross(dVertexNormalW,vObjectSpaceUpW);
```

Comment:
Hmm - yes that's a good point - if the interpolated vertex normal does line up exactly with the object space up vector (which is certainly possible) we will get problems. I can add a check? This will affect performance - but I guess we can assume that meshes set up for GGX anisotropic specular will be able to use one of the more efficient shader chunks - ideally: TBNfastPS?

Reply:
I think a check here will be fine, better than nans.  And as you say, this is hopefully not a common code path.

Reply:
Check added to shader chunk - tested with boxes (as suggested) - BTW it would be good to have versions of the built in primitives available that have tangents for examples that may need them?